### ISSUE TITLE
Send less data to remote transient workers

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#434]
In `make(parallelism = "future")` ([`future.R`](https://github.com/ropensci/drake/blob/master/R/future.R)) each transient worker only needs the imports and the direct target dependencies. The other targets do not need to be sent.

**Comment by wlandau** [comment]
For non-global environments, it may be worth making a deep copy and pruning it aggressively.

**Comment by kendonB** [comment]
Doesn't `future` handle this automatically?

**Comment by wlandau** [comment]
Yes, `future` uses the `globals` package to detect elements of the global environment, but `drake` is using its own dependency detection technique, and the results may disagree.

**Comment by wlandau** [comment]
And because of NFS file latency issues, I would rather not have transient workers read from the `.drake` folder before building their targets. I think we can find a win-win here.

**Comment by wlandau** [comment]
Like many things, this will be much easier after #440.
